File: src/main/java/kr/devport/api/repository/ArticleRepository.java
package kr.devport.api.repository;

import kr.devport.api.domain.entity.Article;
import kr.devport.api.domain.enums.Category;
import kr.devport.api.domain.enums.Source;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ArticleRepository extends JpaRepository<Article, Long> {

    Page<Article> findAll(Pageable pageable);

    Page<Article> findByCategory(Category category, Pageable pageable);

    Page<Article> findBySource(Source source, Pageable pageable);

    Page<Article> findBySourceOrderByScoreDesc(Source source, Pageable pageable);

    List<Article> findAllByOrderByScoreDescCreatedAtSourceDesc(Pageable pageable);
}

File: article/repository-jpa/src/main/kotlin/kr/devport/api/domain/article/repository/ArticleRepository.kt
package kr.devport.api.domain.article.repository

import kr.devport.api.domain.article.entity.Article
import kr.devport.api.domain.article.enums.Category
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface ArticleRepository :
    JpaRepository<Article, Long>,
    ArticleRepositoryCustom {
    fun findByExternalId(externalId: String): Article?

    fun findByCategory(
        category: Category,
        pageable: Pageable,
    ): Page<Article>

    fun findBySource(
        source: String,
        pageable: Pageable,
    ): Page<Article>

    fun findBySourceOrderByScoreDesc(
        source: String,
        pageable: Pageable,
    ): Page<Article>

    fun findAllByOrderByScoreDescCreatedAtSourceDesc(pageable: Pageable): List<Article>
}
